message_definitions/v1.0/development.xml

```diff
@@ -38,5 +44,25 @@
       <field type="float" name="raw_press" units="hPa">Raw differential pressure. NaN for value unknown/not supplied.</field>
       <field type="uint8_t" name="flags" enum="AIRSPEED_SENSOR_FLAGS">Airspeed sensor flags.</field>
     </message>
+    <message id="420" name="RADIO_RC_CHANNELS">
+      <description>RC channel inputs for a flight controller and associated components; the message does not replace the RC_CHANNELS and similar output messages.
+        Supports up to 32 channels. Channel values are in centered 13 bit format. Range is -4096 to 4096, center is 0. Conversion to PWM is x * 5/32 + 1500.
+        The target_system field should normally be set to the system id of the system to control, typically the flight controller.
+        The target_component field can normally be set to 0, so that all components of the system can receive the message.
+        The time_last_update_ms field indicates the time, in the sender's time domain, when the RC channels data were last updated. If this time is significantly shorter
+        than the expected update time for this message, then the RC channels data can be considered invalid, and the RC channels data should carry the last valid data.
```

Comment:
Shorter or longer? 
- I think you're saying if that our last update time gets too big the receiver can know that it should ignore the data. 
- I think the data being invalid would always be decided by the receiver right - or to put it another way the data in this message should always be whatever it last stored that matches the update time.
- If we want the thing sending the message to make the decision about how long is too long then perhaps that's a fault that should be send in the message (maybe what failsafe means?)

Reply:
You make a very valid observation, that with regards to targets and this timestamp quite something has changed here, and I will comment to this big picture below. 

As regards the mechansim, I think it's pretty simple and common (even though maybe not totally trivial to implement). First note that we have 3 situations: 1. frame with valid rc data received, 2. a frame is missed (which isn't equal to a failsafe!), ergo no fresh rc data in that time slot, 3. failsafe. So we need "something" to indicate case 2. In sbus language it would be called "missed frame". Instead of a 2nd flag (which was the original proposal) ,this field was introduced upon @tridge 's request, and is supoposed to help in case of a lossy network. So, the basic is the assumption that not all messages are receiveed but some are lost.

Reply:
Thanks.

>  If this time is significantly shorter than the expected update time for this message, then the RC channels data can be considered invalid, and the RC channels data should carry the last valid data." 

The timestamp would be small if the sender rebooted or similar -  so is this to catch the case where the sender rebooted and you can no longer use the delta between two timestamps to determine the time between the last two updates?